streaming-binlog/src/test/java/org/openmrs/analytics/FhirConverterTest.java

```diff
@@ -92,6 +107,24 @@ public void shouldFetchFhirResourceAndUploadToFhirStore() throws IOException {
 		Mockito.verify(parquetUtil, Mockito.never()).write(Mockito.<Resource> any());
 	}
 	
+	@Test
+	public void shouldGetUuidFromParent() throws SQLException, PropertyVetoException, ClassNotFoundException {
+		
+		Map<String, String> messageBody = DebeziumTestUtil.genExpectedBody();
+		Map<String, Object> messageHeaders = DebeziumTestUtil.genExpectedHeaders(Operation.CREATE, "patient");
+		String uuid = getUuidUtil.getUuid(Mockito.anyString(), Mockito.anyString(), Mockito.anyString());
```

Comment:
Why do you initialize `uuid` like this? Why can't `uuid` be a constant string?

Reply:
@bashir2 the uuid has to be a result of calling the `getUuid() `method which we verify that it has been called and its number of times

Reply:
What are you trying to unit-test here? To me, the production code that is being tested is `sendBodyAndHeaders()` and you want to make sure that the correct `uuid` is fetched from the parent table, correct? If yes, then you can use any arbitrary string as `uuid` (hence the constant) and set the right mock behavior on `uuidUtil`, e.g., 
```
when(getUuidUtil.getUuid(...)).thenReturn(uuid)
```
(the arguments should not be `anyString`, they should be what you expect this function to be called with during `sendBodyAndHeaders()`).
Then, you can also verify that `fetchFhirResource` is called with the right url which includes the above `uuid` (and not just `anyString()`).